internal/providers/github/properties/repository.go

```diff
@@ -128,7 +111,29 @@ func getRepoWrapper(
 
 	repoProps[properties.PropertyName] = fmt.Sprintf("%s/%s", repo.GetOwner().GetLogin(), repo.GetName())
 
-	return repoProps, nil
+	return repoProps
+}
+
+func getRepoWrapper(
+	ctx context.Context, ghCli *go_github.Client, isOrg bool, getByProps *properties.Properties,
+) (map[string]any, error) {
+	_ = isOrg
+
+	name, owner, err := getNameOwnerFromProps(ctx, getByProps)
+	if err != nil {
+		return nil, fmt.Errorf("error getting name and owner from properties: %w", err)
+	}
+	zerolog.Ctx(ctx).Debug().Str("name", name).Str("owner", owner).Msg("Fetching repository")
+
+	repo, result, err := ghCli.Repositories.Get(ctx, owner, name)
+	if err != nil {
+		if result != nil && result.StatusCode == http.StatusNotFound {
+			return nil, v1.ErrEntityNotFound
+		}
+		return nil, err
+	}
+
+	return GitHubRepoToMap(repo), nil
```

Comment:
I can't tell if it would make the code easier or harder, but in my branch I alrady had to add Properties.ToProtoStruct...but I guess here you specifically need to feed the function a github repo?

Reply:
Pretty much.